src/simtools/visualization/simtel_event_plots.py

```diff
@@ -0,0 +1,612 @@
+#!/usr/bin/python3
+"""Plots for light emission (flasher/calibration) sim_telarray events."""
+
+from __future__ import annotations
+
+import logging
+
+import astropy.units as u
+import matplotlib.pyplot as plt
+
+__all__ = [
+    "plot_simtel_event_image",
+    "plot_simtel_integrated_pedestal_image",
+    "plot_simtel_integrated_signal_image",
+    "plot_simtel_peak_timing",
+    "plot_simtel_step_traces",
+    "plot_simtel_time_traces",
+    "plot_simtel_waveform_pcolormesh",
+]
+
+_logger = logging.getLogger(__name__)
+
+# Reusable literal constants (duplicated from visualize to avoid circular deps)
+AXES_FRACTION = "axes fraction"
+NO_R1_WAVEFORMS_MSG = "No R1 waveforms available in event"
+TIME_NS_LABEL = "time [ns]"
+R1_SAMPLES_LABEL = "R1 samples [a.u.]"
+
+
+def _select_event_by_type(source):
+    """Return the first event from the source."""
+    for ev in source:
+        return ev
+    _logger.warning("No events available from source")
+    return None
+
+
+def plot_simtel_event_image(filename, distance=None):
+    """
+    Read in a sim_telarray file and plot DL1 image via ctapipe.
+
+    Returns matplotlib.figure.Figure or None.
+    """
+    # pylint:disable=import-outside-toplevel
```

Comment:
Why imports outside the top level? This is a plotting routine, which will be anyway slow.